src/printSchema.ts

```diff
@@ -49,9 +49,57 @@ export function applySDLHeader(config: GratsConfig, sdl: string): string {
 export function printSDLWithoutMetadata(doc: DocumentNode): string {
   const trimmed = visit(doc, {
     ScalarTypeDefinition(t) {
-      return specifiedScalarTypes.some((scalar) => scalar.name === t.name.value)
-        ? null
-        : t;
+      if (t.isExternalType) {
+        return null;
+      } else if (
+        specifiedScalarTypes.some((scalar) => scalar.name === t.name.value)
+      ) {
+        return null;
+      } else {
+        return t;
+      }
+    },
+    ObjectTypeDefinition(t) {
+      if (t.isExternalType) {
+        return null;
+      } else {
+        return t;
+      }
+    },
+    InterfaceTypeDefinition(t) {
+      if (t.isExternalType) {
+        return null;
+      } else {
+        return t;
+      }
+    },
+    ObjectTypeExtension(t) {
+      if (t.isExternalType) {
+        return null;
+      } else {
+        return t;
+      }
+    },
+    UnionTypeDefinition(t) {
+      if (t.isExternalType) {
+        return null;
+      } else {
+        return t;
+      }
+    },
+    EnumTypeDefinition(t) {
+      if (t.isExternalType) {
+        return null;
+      } else {
+        return t;
+      }
+    },
+    InputObjectTypeDefinition(t) {
+      if (t.isExternalType) {
+        return null;
+      } else {
+        return t;
+      }
```

Comment:
What do you think about implementing this (stripping external types) as its own transform, rather than building it into printing?